File: frontend/app/lib/constants/error-message.ts
export enum HTTP_STATUS_CODE {
  OK = 200,
  CREATED = 201,
  BAD_REQUEST = 400,
  UNAUTHENTICATED = 401,
  FORBIDDEN = 403,
  NOT_FOUND = 404,
  REQUEST_TIMEOUT = 408,
  TOO_MANY_REQUESTS = 429,
  INTERNAL_SERVER_ERROR = 500,
}

export enum API_ERROR_MESSAGE_HEADER {
  UNAUTHENTICATED = "🔐 Access Denied",
  FORBIDDEN = "🚫 Forbidden",
  RATE_LIMIT = "⏱️ Too Many Requests",
  UNEXPECTED_ERROR = "⚠️ Unexpected Error",
}

export enum API_ERROR_MESSAGE {
  ERROR_401_UNAUTHENTICATED = "Please log in to access this page.",
  ERROR_403_FORBIDDEN = "You don’t have the necessary permissions to view this page. If you believe this is an error, please contact support.",
  ERROR_429_RATE_LIMIT_EXCEEDED = "Rate limit exceeded. Please reduce the frequency of your requests and try again later.",
  UNEXPECTED_ERROR = "An unexpected error occured. Please try again later.",
}

export enum CoffeeRecipeErrMsg {
  COFFEE_DOSE_REQUIRED = "Coffee dose is required",
  COFFEE_DOSE_NUMBER = "Coffee dose must be a number",
  COFFEE_DOSE_MIN = "Coffee dose must be at least 15g",
  WATER_AMOUNT_REQUIRED = "Water amount is required",
  WATER_AMOUNT_NUMBER = "Water amount must be a number",
  WATER_AMOUNT_MIN = "Water amount must be at least 10ml",
  BREW_TEMP_REQUIRED = "Brew temperature is required",
  BREW_TEMP_NUMBER = "Brew temperature must be a number",
  BREW_TEMP_MIN = "Brew temperature must be at least 0°C",
  BREW_TIME_REQUIRED = "Brew time is required",
  BREW_INSTRUCTIONS_MIN = "At least one brew step is required",
  BREW_INSTRUCTION_EMPTY = "Each brew step must not be empty",
  CREATED_BY_REQUIRED = "Barista name is required",
}

export enum CoffeeListingErrMsg {
  COFFEE_NAME_REQUIRED = "Coffee name is required",
  ROAST_DATE_REQUIRED = "Roast date is required",
  WEIGHT_REQUIRED = "Weight is required",
  WEIGHT_NUMBER = "Weight must be a number",
  WEIGHT_MIN = "Weight must be greater than 0",
  ROAST_TYPE_REQUIRED = "Roast type is required",
  BREW_METHOD_REQUIRED = "Brew method is required",
}

export enum AIChatMessage {
  MESSAGE_REQUIRED = "Message is required",
  MAX_CHARACTERS_500 = "Message must be under 500 characters",
}

File: backend/src/main/java/com/bi/constant/ErrorMessage.java
package com.bi.constant;

public class ErrorMessage {
    // Authentication
    public static final String UNAUTHORISED_USER = "User is unauthorised. Please login.";
    public static final String USER_ID_REQUIRED = "UserId cannot be null";
    public static final String USER_DOES_NOT_EXIST = "User does not exist";

    // Rate Limiting
    public static final String TOO_MANY_REQUESTS = "Too many requests. Please try again later.";

    // AddOrUpdateCoffeeListingDTO
    public static final String COFFEE_NAME_REQUIRED = "Coffee name is required";
    public static final String ROAST_DATE_REQUIRED = "Roast date is required";
    public static final String NO_FUTURE_DATE = "Date cannot be in the future";
    public static final String WEIGHT_REQUIRED = "Weight is required";
    public static final String POSITIVE_ONLY = "Number should never be zero or negative";
    public static final String ROAST_TYPE_REQUIRED = "Roast type is required";
    public static final String BREW_METHOD_REQUIRED = "Brew method is required";

    // AddOrUpdateRecipeDTO
    public static final String METHOD_TYPE_REQUIRED = "Method type is required";
    public static final String COFFEE_DOSE_REQUIRED = "Coffee dose is required";
    public static final String WATER_AMOUNT_REQUIRED = "Water amount is required";
    public static final String BREW_TEMP_REQUIRED = "Brew temperature is required";
    public static final String BREW_TIME_REQUIRED = "Brew time is required";
    public static final String BREW_INSTRUCTIONS_REQUIRED = "Brew instructions are required";
    public static final String CREATED_BY_REQUIRED = "Creator's name is required";
}
